examples/workflow-test/src/graph/elements/task-automated.po.ts

```diff
@@ -0,0 +1,38 @@
+import {
+    ChildrenAccessor,
+    Clickable,
+    CommandPaletteCapability,
+    EdgesAccessor,
+    Hoverable,
+    MarkerCapability,
+    Mix,
+    PConnectableElement,
+    PParentElement,
+    ResizeHandleCapability,
+    TooltipCapability
+} from '@eclipse-glsp/glsp-playwright-core/extension';
+import { NodeMetadata, PNode, SVGMetadataUtils } from '@eclipse-glsp/glsp-playwright-core/glsp';
+import { LabelHeading } from './label-heading.po';
+
+export const TaskAutomatedMixin = Mix(PNode)
+    .flow(Clickable)
+    .flow(Hoverable)
+    .capability(ResizeHandleCapability)
+    .capability(TooltipCapability)
+    .capability(CommandPaletteCapability)
+    .capability(MarkerCapability)
+    .build();
+
+@NodeMetadata({
+    type: 'task:automated'
+})
+export class TaskAutomated extends TaskAutomatedMixin implements PConnectableElement, PParentElement {
```

Comment:
It all works well and it looks could but it is also a bit hard to understand due to the mixture of mixin super class wrapping a lot of other types and the implementation of additional interfaces. Why do we need to explicitly implement some of those interfaces, e.g., I believe that every PNode is automatically a `PParentElement` if I'm not mistaken.

Reply:
Yep, `PNode` should have accessing `children` and `edges` already built in. 

There are two things: 
- Using Mixins: This allows the developers to reuse functionality that they wish without themselves implementing those
- However, we can not provide a default implementation for everything, some cases require that the developers use those interfaces explicitly and provide the functionality

Reply:
Got it, as soon as we have a bit more documentation I'm sure it is also more straight-forward for everyone else.